qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -438,7 +439,7 @@ def __init__(self, name: str, address: str, silent: bool = False,
         ####################################
         # Instrument specifications
 
-        self.has_DIG = 'DIG' in self._licenses()
+        self.has_DIG = 'DIG' in self._licenses() or LooseVersion('A.03') <= LooseVersion(idn['firmware'])
```

Comment:
Should this also check that the instrument is the correct model? 

So along the lines of 

```
self.has_DIG = 'DIG' in self._licenses() or (LooseVersion('A.03') <= LooseVersion(idn['firmware']) and model in modelswithdig
```
```

Reply:
That could be an option.. At the moment, the self.has_DIG only seems to affect the dicts for certain models so with this implementation it would seem redundant to also specify the model, small example from the code:
```
        PLCs = {'34410A': [0.006, 0.02, 0.06, 0.2, 1, 2, 10, 100],
                '34460A': [0.02, 0.2, 1, 10, 100],
                '34461A': [0.02, 0.2, 1, 10, 100],
                '34465A': [0.02, 0.06, 0.2, 1, 10, 100],
                '34470A': [0.02, 0.06, 0.2, 1, 10, 100]
                }
        if self.has_DIG:
            PLCs['34465A'] = [0.001, 0.002, 0.006] + PLCs['34465A']
            PLCs['34470A'] = [0.001, 0.002, 0.006] + PLCs['34470A']
```

Reply:
Nope it also affects certain options such as the pre trigger. Most places it checks has_DIG and model number but not every where.

Reply:
* https://github.com/QCoDeS/Qcodes/blob/7ca399a05c3c559160b3a4f2a532319f7551d07f/qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py#L87
* https://github.com/QCoDeS/Qcodes/blob/7ca399a05c3c559160b3a4f2a532319f7551d07f/qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py#L123
* https://github.com/QCoDeS/Qcodes/blob/7ca399a05c3c559160b3a4f2a532319f7551d07f/qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py#L175
* https://github.com/QCoDeS/Qcodes/blob/7ca399a05c3c559160b3a4f2a532319f7551d07f/qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py#L363

Reply:
I think that line 175 and 363 will only work for the 34465 and 34470 too. These may now evaluate to true for other models.

Reply:
Yes, this would seem to be the case according to this table: 
![afbeelding](https://user-images.githubusercontent.com/9569658/69710851-65929980-1100-11ea-9c8b-c377ebaf96f4.png)

Reply:
So I would suggest that we change that line to

```python
# 34465A and 34470A supports additional options with the DIG extension
# this is an optional licenced extension with older versions of the firmware
# but is included in the baseline config from version A.03 onward.  
self.has_DIG = ('DIG' in self._licenses() or LooseVersion('A.03') <= LooseVersion(idn['firmware'])) and self.is_34465A_34470A
```

Then we can get rid of the model checks on line 87 and 123.

Reply:
Looking at the table above, the DIG can be either activated via license or the new firmware so that's ok I think. The other models don't have either the option for DIG or MEM, so the model checks are also good. 
Now the DIG activates special triggering and more values for this. The MEM only activates the pretrigger to go up to 2e6 for pretriggering. I will implement this and let's have a second look at this from that point.

Reply:
Now DIG is either enabled by the license or the newer firmware. The MEM will only be enabled if the license is added to the device. This will only enable more counts in the pretrigger. The other changes come with either DIG license or new firmware.

Reply:
Nice!